java-core/google-cloud-core/src/test/java/com/google/cloud/TimestampTest.java

```diff
@@ -237,14 +237,18 @@ void parseTimestampWithoutTimeZoneOffset() {
 
   @Test
   void parseTimestampWithTimeZoneOffset() {
```

Comment:
From discussion with @blakeli0, let's increment this test to cover all possible cases where the `threeten` behavior differs from `java.time`

Reply:
@blakeli0 I expanded the tests to include more edge cases. The tests using `threeten` behave in the same way (see https://github.com/googleapis/sdk-platform-java/pull/3339).
WDYT?

Reply:
Thanks, it LGTM. IIUC, `appendZoneOrOffsetId` does add new things like `Zone` to the formatter, but we don't really use it in `Timestamp`. The parsed `Timestamp` in the end only includes  `seconds` and `nanos` which should not be affected by this change.

Can you please add a few test cases to this PR? For example, test parsing Strings with zone info?

Reply:
Thank you.

> test parsing Strings with zone info?

I added a few in a90bf83b4faa8fd3bf9ff7b1513b6c600a0692e4

https://github.com/googleapis/sdk-platform-java/blob/b67db509d5f121a42b70eec1dd5824358053d973/java-core/google-cloud-core/src/test/java/com/google/cloud/TimestampTest.java#L291-L302

Although we don't expect zone IDs to be parsed, I added a few tests to confirm they behave like the docs explain it would.